docs/defaulting.md

```diff
@@ -0,0 +1,249 @@
+
+## The Problem
+
+ * Hotplug operations are difficult with the minimal VM spec of kubevirt,
+   since we could match wrong devices.
+ * After VM restarts, different devices or addresses could be assigned to a VM
+   by libvirt
+ * libvirt assigns all that data on the host, either when defining the Domain,
+   or after the Domain was started. That requires asynchronous lookups and in
+   the worst case even merge logic.
+ * If we don't want to change Guest OS relevant fields in the VM specification
+   behind the back of the user, all computed default values by libvirt, need to
+   be reflected in the VM status.
+
+If your VMs are ephemeral (no migrations, recreating VMs vs. restarting VMs, no
+hotplug), you are not affected by that.
+
+## Proposed Solution
+
+### Defaulting service based on libvirt
+
+Except for some runtime information, which is normally not included in
+migratable domain XMLs, libvirt fills in all defaults when defining a domain,
+which influence how the emulated machine looks like for the Guest OS. We can
+leverage that with a simple REST based service, which defines VMs in an
+isolated libvirt (running libvirt and the small REST service in a Pod). The
+service is a standalone component in KubeVirt and can be integrated in the
+following flow:
+
+ * When doing a POST to
+   `/apis/kubevirt.io/v1alpha1/namespaces/mynamespace/vms`, `virt-api` can do
+   a roundtrip to that service and prefill the defaults. As the response of
+   the POST to `virt-api`, you will get the completely prefilled VM
+   specification. This updated spec contains all the necessary bits, to allow
+   consistent hotplugging and keeping the VM definition consistent between
+   redefines.
+
+Additional flows, like a dedicated `dry-run` endpoint, to allow management
+applications, reusing existing admission controller logic for prefilling VMs,
+withouth having to actually run a VM are thinkable. That would allow complete
+decoupling of managment and runtime.
+
+Defaulting rules:
+
+ * The Kubevirt VM spec can contain abstract definitions, like a reference to a
+   volume claim, or a reference to a host network. In such cases most of the
+   time, the device `source` section needs to be replaced by a dummy `source`
+   inside the defaulter. Then the fully populated `target` section needs to be
+   mapped back (e.g. a PVC does not have to exist, when posting a VM. It can be
+   mocked out, since it only provides the backend configuration).
+ * Fields which don't influence how the Guest OS sees, shall not be defaulted.
+   Such fields need to be filtered out.
+ * There exist mandatory Domain XML fields, which we can't allow to be set on
+   the VM specification. In such cases, the defaulter needs to set dummy values
+   before asking libvirt for the defaults. Since these fields should not exist
+   on the VM specification at all (see
+   [docs/vm-configuration.md](docs/vm-configuration.md)), not mapping such
+   dummy data back should be easy.
```

Comment:
ok. but you do want to have these expressed in VM spec...so the comment relates just to the defaulting, right?